fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/FHIRResourceDAOFactory.java

```diff
@@ -7,77 +7,71 @@
+     * @param trxSynchRegistry
+     * @return a concrete implementation of {@link ResourceDAO}
+     * @throws IllegalArgumentException
+     * @throws FHIRPersistenceException
+     */
+    public static ResourceDAO getResourceDAO(Connection connection, String schemaName, FHIRDbFlavor flavor, TransactionSynchronizationRegistry trxSynchRegistry)
+        throws IllegalArgumentException, FHIRPersistenceException {
         ResourceDAO resourceDAO = null;
-        switch (FHIRResourceDAOType.of(conn)) {
-            case DB2:
-                resourceDAO = new ResourceDAOImpl(trxSynchRegistry);
-                break;
-            case DERBY:
-                resourceDAO = new DerbyResourceDAO(trxSynchRegistry);
-                break;
-            case POSTGRESQL:
-                resourceDAO = new PostgreSqlResourceDAO(trxSynchRegistry);
-                break;
+        
+        switch (flavor.getType()) {
+        case DB2:
+            resourceDAO = new ResourceDAOImpl(connection, schemaName, flavor, trxSynchRegistry);
+            break;
+        case DERBY:
+            resourceDAO = new DerbyResourceDAO(connection, schemaName, flavor, trxSynchRegistry);
+            break;
+        case POSTGRESQL:
+            resourceDAO = new PostgreSqlResourceDAO(connection, schemaName, flavor, trxSynchRegistry);
+            break;
         }
         return resourceDAO;
     }
 
-    public static ResourceDAO getResourceDAO (Connection conn) throws IllegalArgumentException, SQLException {
+    /**
+     * Construct a new ResourceDAO implementation matching the database type
+     * @param connection valid connection to the database
+     * @param schemaName the name of the schema containing the FHIR resource tables
+     * @param flavor the type and capability of the database and schema
+     * @return a concrete implementation of {@link ResourceDAO}
+     * @throws IllegalArgumentException
+     * @throws FHIRPersistenceException
+     */
+    public static ResourceDAO getResourceDAO(Connection connection, String schemaName, FHIRDbFlavor flavor) throws IllegalArgumentException, FHIRPersistenceException {
         ResourceDAO resourceDAO = null;
-        switch (FHIRResourceDAOType.of(conn)) {
-            case DB2:
-                resourceDAO = new ResourceDAOImpl(conn);
-                break;
-            case DERBY:
-                resourceDAO = new DerbyResourceDAO(conn);
-                break;
-            case POSTGRESQL:
-                resourceDAO = new PostgreSqlResourceDAO(conn);
-                break;
+        switch (flavor.getType()) {
+        case DB2:
+            resourceDAO = new ResourceDAOImpl(connection, schemaName, flavor);
+            break;
+        case DERBY:
+            resourceDAO = new DerbyResourceDAO(connection, schemaName, flavor);
+            break;
+        case POSTGRESQL:
+            resourceDAO = new PostgreSqlResourceDAO(connection, schemaName, flavor);
+            break;
```

Comment:
default behavior?  

ps. seems very duplicate of prior method?

Reply:
No default required.

One method passed a trxSynchRegistry.